### DISCUSSION TITLE
Possible to apply default security filter on given index?

### DISCUSSION THREAD

**Discussion by landermedia** [algolia/instantsearch#5472] [Q&A] [answered]
According to [User-restricted access to data](https://www.algolia.com/doc/guides/security/api-keys/how-to/user-restricted-access-to-data/) guide, you can limit to a subset of data but that means you have to generate this secured API key and use it to make it work. 

Is it possible to apply the given filter `filters: `visible_by:${currentUserID}` by default in index setting level?

**Comment by Haroenv** [comment, marked as answer]
You wouldn't know what that `currentUserID` is in the settings, as that is only available in your frontend/backend logic. You should instead generate a secured api key when the user is loading the page, and use that for InstantSearch

**Reply by landermedia** [reply]
My bad, you are correct but let's say we have a value for that by default. That's why we're asking if it's possible to have such filter setting applied without having to generate a search key? I don't seem to find it in `index.setSettings` options so I doubt it's possible or maybe I'm missing something.

**Reply by Haroenv** [reply]
You can create an api key statically with filters applied too. In the dashboard creating a new api key has the option for a filter attached, which users also won't be able to remove.

**Reply by landermedia** [reply]
Thanks @Haroenv. I guess you're referring to the query parameters?

<img width="616" alt="image" src="https://user-images.githubusercontent.com/83575285/217266725-2be3882e-efb6-4927-b1b6-95c87ebe9f2d.png">

**Reply by Haroenv** [reply]
Yes, that's equivalent :)

**Reply by landermedia** [reply]
Thank you @Haroenv 😊
